test/integration/content-explorer/ContentExplorer.e2e.test.js

```diff
@@ -0,0 +1,201 @@
+import localize from '../../support/i18n';
+
+// <reference types="Cypress" />
+const selectedRowClassName = 'bce-item-row-selected';
+const selectedRowClassSelector = `.${selectedRowClassName}`;
+const helpers = {
+    load(additionalProps = {}) {
+        cy.visit('/Elements/ContentExplorer', {
+            onBeforeLoad: contentWindow => {
+                contentWindow.PROPS = additionalProps;
+            },
+        });
+    },
+    // cy.contains('Upload') returns the 'canUpload' help text
+    // at the bottom of the page, as opposed to the Upload button.
+    // Use this function when you need cy.contains() to match exact text.
+    getExactRegex(str) {
+        return new RegExp(`^${str}$`);
+    },
+    getRow(rowNum) {
+        return cy.getByTestId('content-explorer').find(`.bce-item-row-${rowNum}`);
+    },
+    checkRowSelections(selectedRow) {
+        if (selectedRow) {
+            cy.getByTestId('content-explorer')
+                .find(`${selectedRowClassSelector}`)
+                .should('have.length', 1)
+                .should('have.class', `bce-item-row-${selectedRow}`);
+        } else {
+            cy.getByTestId('content-explorer').should('not.have.descendants', selectedRowClassSelector);
+        }
+    },
+    selectRow(rowNum) {
+        this.getRow(rowNum)
+            .as('row')
+            .click()
+            .should('have.class', selectedRowClassName);
+        return cy.get('@row');
+    },
+    getAddButton() {
+        return cy.getByTestId('be-btn-add');
+    },
+    getUploadButton() {
+        return cy.contains(this.getExactRegex(localize('be.upload')));
+    },
+    getNewFolderButton() {
+        return cy.contains(localize('be.newFolder'));
+    },
+    getCancelButton() {
+        return cy.contains(localize('be.cancel'));
+    },
+    getCloseButton() {
+        return cy.contains(localize('be.close'));
+    },
+    getShareButton(rowNum) {
+        return this.getRow(rowNum).contains(localize('be.share'));
+    },
+    getMoreOptionsButton(rowNum) {
+        return this.getRow(rowNum).find('[data-testid="bce-btn-more-options"]');
+    },
+    getRenameButton() {
+        return cy.contains(this.getExactRegex(localize('be.rename')));
+    },
+    getItemNameFromRow(rowNum) {
+        return this.getRow(rowNum).find('[data-testid="be-item-name"]');
+    },
+    getClosePreviewButton() {
+        return cy.getByTestId('bcpr-btn-close');
+    },
+    openUploadModal() {
+        this.getAddButton().click();
+        this.getUploadButton().click();
+    },
+    openNewFolderModal() {
+        this.getAddButton().click();
+        this.getNewFolderButton().click();
+    },
+    previewItemFromRow(rowNum) {
+        this.getItemNameFromRow(rowNum).click();
+    },
+};
+
+describe('ContentExplorer', () => {
+    beforeEach(() => {
+        cy.server();
+        cy.route('GET', '**/folders/*', 'fixture:content-explorer/root-folder.json');
```

Comment:
We should probably rename this file to `ContentExplorer.int.test.js`, since its backend is stubbed out.